bson/src/test/unit/org/bson/codecs/pojo/PojoTestCase.java

```diff
@@ -90,8 +90,12 @@ <T> void roundTrip(final T value, final String json) {
     }
 
     <T> void roundTrip(final PojoCodecProvider.Builder builder, final T value, final String json) {
-        encodesTo(getCodecRegistry(builder), value, json);
-        decodesTo(getCodecRegistry(builder), json, value);
+        roundTrip(builder, value.getClass(), value, json);
+    }
+
+    <T> void roundTrip(final PojoCodecProvider.Builder builder, final Class<?> clazz, final T value, final String json) {
```

Comment:
This is needed so the class of the Codec can be different from the class of the entity, as it is in the regression described on the JAVA ticket.